lib/runtime/test_helpers.go

```diff
@@ -85,8 +85,7 @@ func GetRuntimeBlob(testRuntimeFilePath, testRuntimeURL string) error {
 		return nil
 	}
 
-	/* #nosec */
-	resp, err := http.DefaultClient.Get(testRuntimeURL)
+	resp, err := http.Get(testRuntimeURL)
```

Comment:
That uses the DefaultClient as well 😄 You should create an http client in your helper function e.g. `client := &http.Client{Timeout: time.Second}`. Ideally we would have a global HTTP client for tests so connections are re-used, but since `test_helpers.go` does not end with `_test.go`, its code might be incorporated in the production binary so let's not do that.

Two things regarding get requests:

- We should use a custom http client (even for tests)
- We should use the `http.NewRequestWithContext` followed by `client.Do(request)` instead to have cancelable

Reply:
got it! implemented